src/lib/transition_frontier/persistent_frontier/worker.ml

```diff
@@ -126,7 +130,11 @@ module Worker = struct
             Ok ()
         | _ ->
             map_error ~diff_type:`New_node ~diff_type_name:"New_node" r )
-    | Root_transitioned {new_root; garbage= Lite garbage} ->
+    | Root_transitioned {new_root; garbage= Lite garbage; just_emitted_a_proof}
+      ->
+        if just_emitted_a_proof then
+          Persistent_root.Instance.dequeue_snarked_ledger
```

Comment:
Can we sanity check the hash of the ledger we are dequeuing against the snarked ledger hash of new root's protocol state in the db?

Reply:
Wouldn't it be equal to the ledger hash of the old root? from what I understand the oldest checkpoint corresponds to the persisted frontier root

Reply:
Err, yes, I think that's correct. The new root's ledger hash should be the checkpoint at the front of the queue after the dequeue, not the one we are dequeuing.